internal/jujuapi/httpproxy.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2024 Canonical.
+package jujuapi
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/juju/names/v5"
+	"github.com/juju/zaputil/zapctx"
+	"go.uber.org/zap"
+
+	"github.com/canonical/jimm/v3/internal/rpc"
+)
+
+type HTTPProxier struct {
+	JIMM JIMM // interface
+}
+
+// ServeHTTP extract the model uuid from the path to retrieve necessary information to proxy the request to the right controller.
+func (s *HTTPProxier) ServeHTTP(ctx context.Context, w http.ResponseWriter, req *http.Request) {
+	writeError := func(msg string, code int) {
```

Comment:
This will be a consistent error message we can use, let's not make an inline function for something that can be obviously used by any handler